packages/core/test/actorLogic.test.ts

```diff
@@ -1032,3 +1273,25 @@ describe('composable actor logic', () => {
     );
   });
 });
+
+interface Deferred<T> {
+  resolve(value: T): void;
+  reject(reason: any): void;
+  promise: Promise<T>;
+}
+
+function defer<T>(): Deferred<T> {
```

Comment:
You should already have this available here at type-level (but we still have to shim the runtime)
```suggestion
function withResolvers<T>(): PromiseWithResolvers<T> {
```